### PR TITLE
Bump cosign e2e tests to rc2.0.0.rc.1

### PR DESCRIPTION
**Pull request by hectorj2f**
#### Summary

continuation of https://github.com/sigstore/policy-controller/pull/535

### CODE REVIEW

File: test/testdata/policy-controller/e2e/cip-key-signature-pull-secrets.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2022 The Sigstore Authors.
```
Comment (cpanato) [inline test/testdata/policy-controller/e2e/cip-key-signature-pull-secrets.yaml:1]:
```suggestion
# Copyright 2023 The Sigstore Authors.
```

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -37,7 +37,7 @@ jobs:
 
       - uses: sigstore/cosign-installer@4079ad3567a89f68395480299c77e40170430341
         with:
-          cosign-release: 'v2.0.0-rc.0'
+          cosign-release: 'v2.0.0-rc.1' # optional
```
Comment (cpanato) [inline .github/workflows/build.yaml:40]: why the comment?

File: test/testdata/policy-controller/e2e/cip-key-signature-pull-secrets.yaml (hunk 1 above)
Comment (hectorj2f) [reply test/testdata/policy-controller/e2e/cip-key-signature-pull-secrets.yaml]: Changed it!

File: .github/workflows/build.yaml (hunk 2 above)
Comment (hectorj2f) [reply .github/workflows/build.yaml]: I am not sure the reasons, tbh. Let's remove it.

**Comment by vaikas** [comment]
I don't think we should be bumping the copyrights in all the files, at least everywhere else they get put in when the file is created and they don't get bumped each year.

File: test/e2e_test_cluster_image_policy_with_tsa.sh [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 #
-# Copyright 2022 The Sigstore Authors.
+# Copyright 2023 The Sigstore Authors.
```
Comment (priyawadhwa) [inline test/e2e_test_cluster_image_policy_with_tsa.sh:3]: nit: this should stay 2022

**Comment by hectorj2f** [comment]
> I don't think we should be bumping the copyrights in all the files, at least everywhere else they get put in when the file is created and they don't get bumped each year.

I dropped the commit!

File: test/e2e_test_cluster_image_policy_with_tsa.sh [hunk 4]
```diff
@@ -171,13 +171,13 @@ echo '::endgroup::'
 # Sign it with key
 echo '::group:: Sign demoimage2 with key, and add to rekor and TSA'
 export TSA_URL=`kubectl -n tsa-system get ksvc tsa -ojsonpath='{.status.url}'`
-COSIGN_EXPERIMENTAL=1 COSIGN_PASSWORD="" cosign sign --key cosign.key --allow-insecure-registry --rekor-url ${REKOR_URL} --timestamp-server-url ${TSA_URL} ${demoimage2}
+COSIGN_EXPERIMENTAL=1 COSIGN_YES="true" COSIGN_PASSWORD="" cosign sign --key cosign.key --allow-insecure-registry --rekor-url ${REKOR_URL} --timestamp-server-url ${TSA_URL} ${demoimage2}
```
Comment (priyawadhwa) [inline test/e2e_test_cluster_image_policy_with_tsa.sh:174]: Maybe remove `COSIGN_EXPERIMENTAL` everywhere since it doesn't do anything anymore?

**Review by priyawadhwa — APPROVED** [review]
looks great! just a couple nits

File: test/e2e_test_cluster_image_policy_with_tsa.sh (hunk 4 above)
Comment (hectorj2f) [reply test/e2e_test_cluster_image_policy_with_tsa.sh:174]: Yeah, that will be done in a separate PR.

Comment (hectorj2f) [reply test/e2e_test_cluster_image_policy_with_tsa.sh:174]: https://github.com/sigstore/policy-controller/pull/535#issuecomment-1407472531

**Review by priyawadhwa — APPROVED** [review]

**Review by vaikas — APPROVED** [review]
Thanks for doing this!

**Pull request merged**